### PR TITLE
Remove Option from window frame 

### PR DESCRIPTION
**Pull request by mustafasrepo**
# Which issue does this PR close?

Closes #.

# Rationale for this change
After parsing the window expression. If window frame is `None`, we can construct its equivalent (in terms of behavior).  
For instance when someone enters the query `OVER(ORDER BY a)`. It is equivalent to `OVER(ORDER BY a) RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW`. Similarly, for the query `OVER(PARTITION BY a)` its equivalent is `OVER(PARTITION BY a ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING)`.

After parsing the necessary information, we can construct the corresponding window frame. Hence in the downstream pipeline we don't need to use Option for `window frame`. 

# What changes are included in this PR?
This PR constructs an equivalent window frame when window frame is `None` after sql parsing. And at the rest of the pipeline uses equivalent window frame. 
Also after equivalent window frame construction default name of the window result column changes. Hence this PR includes changes in the column names of the expected results (I have added aliases to make tests future-proof also). Large bulk of the changes comes from these changes.

# Are these changes tested?

Existing tests should work

# Are there any user-facing changes?

 `api change`

### CODE REVIEW

**Comment by ozankabak** [comment]
This is mostly a refactor PR that simplifies downstream window-processing code by handling empty (or default) window frames only once at the beginning.

I already reviewed it and it LGTM. Let us know if there are any concerns; we will be happy to address them and/or improve the code further.

**Review by alamb — APPROVED** [review]
This is a nice change -- thank you @mustafasrepo and @metesynnada 

I found https://github.com/apache/arrow-datafusion/pull/4516/files?w=1 easier to review


cc @Ted-Jiang  as I believe you are working on window function related things in Ballista

File: datafusion/expr/src/window_frame.rs [hunk 1]
```diff
@@ -66,33 +66,51 @@ impl TryFrom<ast::WindowFrame> for WindowFrame {
...
-impl Default for WindowFrame {
-    fn default() -> Self {
-        WindowFrame {
-            units: WindowFrameUnits::Range,
-            start_bound: WindowFrameBound::Preceding(ScalarValue::Utf8(None)),
-            end_bound: WindowFrameBound::CurrentRow,
+impl WindowFrame {
+    /// Creates a new, default window frame (with the meaning of default depending on whether the
+    /// frame contains an `ORDER BY` clause.
+    pub fn new(has_order_by: bool) -> Self {
```
Comment (Ted-Jiang) [inline datafusion/expr/src/window_frame.rs:95]: Thanks for this fix, i also want to change this default behavior.

File: datafusion/proto/proto/datafusion.proto [hunk 2]
```diff
@@ -573,9 +573,7 @@ message WindowExprNode {
   repeated LogicalExprNode partition_by = 5;
   repeated LogicalExprNode order_by = 6;
   // repeated LogicalExprNode filter = 7;
-  oneof window_frame {
-    WindowFrame frame = 8;
-  }
+  WindowFrame window_frame = 8;
```
Comment (Ted-Jiang) [inline datafusion/proto/proto/datafusion.proto:576]: 👍

**Pull request merged**